### PR TITLE
Actually create the conversation on compose

### PR DESCRIPTION
**Pull request by shichongrui**
refs: MBL-MBL-13788
affects: Parent
release note: none

Test Plan:
 - Create a conversation through the compose workflow
 - The send button should be replaced with an activity
   indicator while it is creating
 - Compose modal should dismiss and the new conversation should
   be in the list
 - NOTE: If you pull to refresh it will go away because we are
   not including sent conversations in the list yet

### CODE REVIEW

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Build failed, skipping coverage check
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Parent

[MBL-13788](https://instructure.atlassian.net/browse/MBL-13788)

<details><summary>:x: Swift lint</summary>


    ⚠️  /Users/vagrant/git/Parent/Parent/Conversations/ComposeViewController.swift:118:151: Unused Closure Parameter Violation: Unused parameter "conversations" in a closure should be replaced with _. (unused_closure_parameter)
    Linting 'TableSectionHeaderView.swift' (77/102)
    Linting 'TokenLabelView.swift' (78/102)
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1018461&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|23
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

File: Core/Core/Conversations/APIConversation.swift [hunk 1]
```diff
@@ -248,3 +248,23 @@ public struct PostAddMessageRequest: APIRequestable {
     public var path: String { "conversations/\(conversationID)/add_message" }
     public let method = APIMethod.post
 }
+
+public struct PostConversationRequest: APIRequestable {
+    public typealias Response = [APIConversation]
```
Comment (armstrongnate) [inline Core/Core/Conversations/APIConversation.swift]: I'm surprised that the response is an array.

File: Parent/Parent/Conversations/ConversationListViewController.swift [hunk 2]
```diff
@@ -100,10 +100,10 @@ class ConversationListViewController: UIViewController, ConversationCoursesActio
     func courseSelected(course: Course, user: User) {
         env.router.route(
             to: .compose(
-                    context: ContextModel(.course, id: course.id),
+                    context: course,
                     observeeID: user.id,
                     subject: course.name,
-                    hiddenMessage: String.localizedStringWithFormat(NSLocalizedString("Regarding: ", bundle: .parent, comment: ""), user.name)
+                    hiddenMessage: String.localizedStringWithFormat(NSLocalizedString("Regarding: %@", bundle: .parent, comment: ""), user.name)
```
Comment (armstrongnate) [inline Parent/Parent/Conversations/ConversationListViewController.swift]: lol nice catch

File: Parent/ParentUnitTests/Conversations/ComposeViewControllerTests.swift [hunk 3]
```diff
@@ -56,8 +56,34 @@ class ComposeViewControllerTests: ParentTestCase {
...
+            body: controller.body(),
+            recipients: [ APIConversationRecipient.make().id.value ],
+            context_code: controller.context!.canvasContextID)
+            ), value: [ APIConversation.make() ]
+        )
+        task.paused = true
         let sendButton = controller.navigationItem.rightBarButtonItem
         XCTAssertNoThrow(sendButton?.target?.perform(sendButton?.action))
+        XCTAssert(controller.navigationItem.rightBarButtonItem?.customView is UIActivityIndicatorView)
+        task.resume()
```
Comment (armstrongnate) [inline Parent/ParentUnitTests/Conversations/ComposeViewControllerTests.swift]: Nice tests!

File: Core/Core/Conversations/APIConversation.swift (hunk 1 above)
Comment (shichongrui) [reply Core/Core/Conversations/APIConversation.swift]: Yep that threw me off at first as well.

File: Core/Core/Conversations/APIConversation.swift [hunk 4]
```diff
@@ -248,3 +248,23 @@ public struct PostAddMessageRequest: APIRequestable {
...
+
+    public struct Body: Encodable {
+        public let subject: String
+        public let body: String
+        public let recipients: [String]
+        public let context_code: String?
+        public let media_comment_id: String? = nil
+        public let media_comment_type: MediaCommentType? = nil
+        public let attachment_ids: [String]? = nil
+        public let group_conversation: Bool? = true
```
Comment (shichongrui) [inline Core/Core/Conversations/APIConversation.swift]: @armstrongnate the request is already defaulting `group_conversation = true`

File: Core/Core/Conversations/APIConversation.swift (hunk 1 above)
Comment (shichongrui) [reply Core/Core/Conversations/APIConversation.swift]: I can add some comments for this.

File: Core/Core/Conversations/APIConversation.swift (hunk 4 above)
Comment (shichongrui) [reply Core/Core/Conversations/APIConversation.swift]: Correct. I will take a look at that.

Comment (armstrongnate) [reply Core/Core/Conversations/APIConversation.swift]: These Jan 15 conversations were all from the same message.

IMG_B13829831E3B-1

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1018631&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|23
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1018633&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|23
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1018783&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|23
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Pull request merged**